### PR TITLE
Test unstake

### LINKED ISSUE #207: Tests for unstake command

### PR DESCRIPTION
**Pull request by Yashk767**
# Description

- Seperated unstake function into initialiseUnstake, executeUnstake and unstake
- Tests for unstake command.

Fixes #207

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: cmd/unstake_test.go [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+	"github.com/ethereum/go-ethereum/common"
+	Types "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/spf13/pflag"
+	"math/big"
+	"razor/core/types"
+	"testing"
+)
+
+//func TestAutoWithdraw(t *testing.T) {
```
Comment (ashish10677) [inline cmd/unstake_test.go:19]: Why is this commented out?

File: cmd/unstake.go [hunk 2]
```diff
@@ -19,74 +22,95 @@ var unstakeCmd = &cobra.Command{
...
+	txnOptions, err := utilsStruct.cmdUtils.Unstake(config, client, address, password, valueInWei, stakerId, utilsStruct)
+	utils.CheckError("Unstake Error: ", err)
+
+	if autoWithdraw {
+		utilsStruct.cmdUtils.AutoWithdraw(txnOptions, stakerId, utilsStruct)
 	}
+}
 
-	stakeManager := utils.GetStakeManager(txnArgs.Client)
+func Unstake(config types.Configurations, client *ethclient.Client, address string, password string, valueInWei *big.Int, stakerId uint32, utilsStruct UtilsStruct) (types.TransactionOptions, error) {
```
Comment (ashish10677) [inline cmd/unstake.go:75]: Can you pass these details which you get from the flags as a single struct.
```
type struct data {
    address string
    password string
    valueInWei *big.Int
    stakerId uint32
}
```

File: cmd/unstake_test.go [hunk 3]
```diff
@@ -0,0 +1,393 @@
...
+	tests := []struct {
+		name    string
+		args    args
+		wantErr error
+	}{
+		{
+			name: "Test 1: When unstake function executes successfully",
+			args: args{
+				lock: types.Locks{
+					Amount: big.NewInt(0),
```
Comment (ashish10677) [inline cmd/unstake_test.go:71]: Does the amount have to be 0 here? If no, then put a realistic value.

File: cmd/unstake_test.go [hunk 4]
```diff
@@ -0,0 +1,393 @@
...
+				value:        big.NewInt(10000),
+				stakerId:     1,
+				lock: types.Locks{
+					Amount: big.NewInt(1000),
+				},
+				unstakeErr: nil,
+			},
+			expectedFatal: true,
+		},
+		//{
```
Comment (ashish10677) [inline cmd/unstake_test.go:297]: If this test isn't required then delete these lines.

**Review by ashish10677 — CHANGES_REQUESTED** [review]

File: cmd/unstake_test.go (hunk 1 above)
Comment (Yashk767) [reply cmd/unstake_test.go:19]: There is good amount of wait time in AutoWithdraw thus its not being tested

File: cmd/unstake_test.go (hunk 3 above)
Comment (Yashk767) [reply cmd/unstake_test.go:71]: Yes it should be 0 as we have a check of existing lock for unstake, If there is existing one than lock.Amount >0.

Comment (ashish10677) [reply cmd/unstake_test.go:71]: Okay 👍

File: cmd/unstake_test.go (hunk 1 above)
Comment (ashish10677) [reply cmd/unstake_test.go]: @SkandaBhat we need to discuss about this, what should be done.

Comment (SkandaBhat) [reply cmd/unstake_test.go]: What is this?

Comment (ashish10677) [reply cmd/unstake_test.go]: While testing unstake and autowithdraw, we need to wait for the lock period to get over. This increases the overall time taken to test the unstake function. What should we do about that?

Comment (Yashk767) [reply cmd/unstake_test.go]: As discussed, I have created mock for sleep function, and added tests for AutoWithdraw.

**Review by ashish10677 — APPROVED** [review]

**Review by SkandaBhat — APPROVED** [review]

**Pull request merged**